src/commands/ext-dev-unpublish.ts

```diff
@@ -0,0 +1,47 @@
+import { Command } from "../command";
+import { logPrefix } from "../extensions/extensionsHelper";
+import { unpublishExtension, parseRef, getExtension } from "../extensions/extensionsApi";
+import * as utils from "../utils";
+import { promptOnce } from "../prompt";
+import * as clc from "cli-color";
+import { requireAuth } from "../requireAuth";
+
+module.exports = new Command("ext:dev:unpublish [ref]")
+  .description("unpublish an extension")
+  .help(
+    "use this command to unpublish an extension, and make it unavailable for developers to install or reconfigure. " +
+      "Specify the extension you want to unpublish using the format '<publisherId/extensionId>'."
+  )
+  .before(requireAuth)
+  .action(async (ref: string, options: any) => {
+    const { publisherId, extensionId, version } = parseRef(ref);
```

Comment:
`parseRef` seems like something that could be used in `:publish` too, if the input parameter needed to change.